server/models/User.ts

```diff
@@ -562,21 +566,9 @@ class User extends ParanoidModel<
 
     if (res.count >= 1) {
       if (to === UserRole.Member) {
-        await this.update(
-          {
-            isAdmin: false,
-            isViewer: false,
-          },
-          options
-        );
+        await this.update({ role: to }, options);
```

Comment:
I'm sure this can all be simplified further now, but doing the minimum to move this forward